### ISSUE TITLE
[Bug]: npx storybook tries to install incorrect version

### ISSUE THREAD

**Issue by rvanheesostk** [storybookjs/storybook#22747]
### Describe the bug

When running `npx storybook@7.0.14 --type react --package-manager=yarn`, the script is trying to install dependencies @7.0.17. Other than the message that this version does not seem to exist, it's not the version I asked for.

Looking at the `./storybook.log`, it's also trying to do it for dependencies other than `@storybook/preview-api` mentioned below.

Full logs below:

```sh
xxxxxxxx@X-XXXXXXXXXX xxxxxxxxxx-xxxx % npx storybook@7.0.14 init --type react --package-manager=yarn

 storybook init - the simplest way to add a Storybook to your project. 

 • Installing Storybook for user specified project type: react. ✓
✔ We were not able to detect the right builder for your project. Please select one: › Webpack 5
 • Adding Storybook support to your "React" app
  ✔ Getting the correct version of 10 packages
✔ We have detected that you're using ESLint. Storybook provides a plugin that gives the best experience with Storybook and helps follow best practices: https://github.com/storybookjs/eslint-plugin-storybook#readme
Would you like to install it? … no
  ⠇ Installing Storybook dependencies
An error occurred while installing dependencies:
YARN1 error: Couldn't find any versions for "@storybook/preview-api" that matches "7.0.17"

Please check the logfile generated at ./storybook.log for troubleshooting and try again.

     Error: YARN1 error: Couldn't find any versions for "@storybook/preview-api" that matches "7.0.17"

Please check the logfile generated at ./storybook.log for troubleshooting and try again.
    at Yarn1Proxy.addDependencies (/Users/xxxxxxxx/.npm/_npx/756d877e324f3eea/node_modules/@storybook/cli/dist/generate.js:40:75)
    at async stopIfExiting (/Users/xxxxxxxx/.npm/_npx/756d877e324f3eea/node_modules/@storybook/cli/dist/generate.js:58:2160)
    at async baseGenerator (/Users/xxxxxxxx/.npm/_npx/756d877e324f3eea/node_modules/@storybook/cli/dist/generate.js:62:2192)
    at async generator3 (/Users/xxxxxxxx/.npm/_npx/756d877e324f3eea/node_modules/@storybook/cli/dist/generate.js:66:5522)
    at async installStorybook (/Users/xxxxxxxx/.npm/_npx/756d877e324f3eea/node_modules/@storybook/cli/dist/generate.js:83:789)
    at async doInitiate (/Users/xxxxxxxx/.npm/_npx/756d877e324f3eea/node_modules/@storybook/cli/dist/generate.js:88:772)
    at async withTelemetry (/Users/xxxxxxxx/.npm/_npx/756d877e324f3eea/node_modules/@storybook/core-server/dist/index.js:35:3620)
    at async initiate (/Users/xxxxxxxx/.npm/_npx/756d877e324f3eea/node_modules/@storybook/cli/dist/generate.js:99:108)
  ⠸ Installing Storybook dependencies%      
```

### To Reproduce

Run `npx storybook@7.0.14 --type react --package-manager=yarn`

### System

```shell
Environment Info:

  System:
    OS: macOS 13.3.1
    CPU: (10) arm64 Apple M1 Pro
  Binaries:
    Node: 16.19.0 - ~/.nvm/versions/node/v16.19.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.19.0/bin/yarn
    npm: 9.4.1 - ~/.nvm/versions/node/v16.19.0/bin/npm
  Browsers:
    Chrome: 113.0.5672.126
    Safari: 16.4
```

**Comment by rvanheesostk** [comment]
Just to add: `@latest` will also try to install `7.0.17`.

**Comment by rvanheesostk** [comment]
Small update on above: The 'missing' version is due to a mismatch in our registry. Downloading directly does work, so that can be ignored.

The issue of trying to install an incorrect version still stands however.

**Comment by vanessayuenn** [comment]
We've fixed this for Storybook @ 7.6+, please upgrade and try it out!
